lib/rubocop/cop/mixin/ignored_methods.rb

```diff
@@ -1,52 +0,0 @@
-# frozen_string_literal: true
-
-module RuboCop
-  module Cop
-    # This module encapsulates the ability to ignore certain methods when
-    # parsing.
-    # Cops that use `IgnoredMethods` can accept either strings or regexes to match
-    # against.
-    module IgnoredMethods
-      # Configuration for IgnoredMethods. It is added to classes that include
-      # the module so that configuration can be set using the `ignored_methods`
-      # class macro.
-      module Config
-        attr_accessor :deprecated_key
-
-        def ignored_methods(**config)
-          self.deprecated_key = config[:deprecated_key]
-        end
-      end
-
-      def self.included(base)
-        base.extend(Config)
-      end
-
-      def ignored_method?(name)
-        ignored_methods.any? do |value|
-          case value
-          when Regexp
-            value.match? String(name)
-          else
-            value == String(name)
-          end
```

Comment:
Because of the processing for regex here, we decided against simply replacing it with `AllowedMethods`.